.tekton/determine-image-digest.yaml

```diff
@@ -0,0 +1,37 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: determine-image-digest
+  namespace: rh-acs-tenant
+spec:
+  description: Determines image digests.
+  params:
+  - name: IMAGE_REPOSITORY
+    description: Image repository.
+    type: string
+  - name: IMAGE_TAG
+    description: Image tag.
+    type: string
+  results:
+  - name: IMAGE_DIGEST
+    description: Image digest in the format `sha256:abcdef0123`.
+  workspaces:
+  - name: pull-secret
+    readOnly: true
+  steps:
+  - name: determine-image-digest
+    image: registry.access.redhat.com/ubi9/skopeo@sha256:a096332b3a6665b1f1e0b34980406649920cf4babba80aff24a715b76f3f62f3
+    env:
+    - name: REGISTRY_AUTH_FILE
+      value: "$(workspaces.pull-secret.path)/.dockerconfigjson"
+    script: |
+      #!/usr/bin/env bash
+      set -euo pipefail
+      image_ref="docker://$(params.IMAGE_REPOSITORY):$(params.IMAGE_TAG)"
+      digest=$(skopeo inspect \
+        --override-os linux \
+        --override-arch amd64 \
```

Comment:
does this limit the architectures that the operator can run on? (I admit I am not very familiar with multi-arch builds for the operator, so this could be a dumb question)

Reply:
These specific parameters (os & arch) are there due to a UX shortcoming of skopeo -- at least that is my understanding.
What we actually need at this point is the digest of the "manifest image", which can be regarded as a container for the platform-specific images.

When I tested these commands locally on my machine (ARM), this skopeo command was failing, because the image was not available for ARM, even though I am only interested in the digest for the manifest image. Providing these parameters made it work -- and what you get from that command (which is probably not very obvious) is actually the digest we need, the one from the manifest image, not for the linux/amd64 specific image.

This is how the parameters ended up in that Tekton task. tbh, I don't know if we should keep them or remove them.

Reply:
Thanks for the explanation, it sounds like it's working as desired.  Perhaps a comment explaining that these overrides are just for local development would help clarify.